.github/workflows/check.yml

```diff
@@ -1,43 +1,87 @@
 name: Check
 
-# XXX disabled for now because we use Jenkins still, but at the time this was tested it was functional
-# on:
-#   pull_request:
-#     branches: [ master ]
+on:
+  pull_request:
+    branches:
+      - '*'  # Trigger on all branches for pull requests
+  push:
+    branches:
+      - master  # Trigger on pushes to the master branch
 
 jobs:
-  check:
+  check_pull_request:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v3
+      - name: Checkout code
+        uses: actions/checkout@v3
 
-    # Note: Actually we don't need Maven CLI, only Java 17, but this may have additional benefits due to the Maven cache
-    - name: Setup Maven
-      uses: s4u/setup-maven-action@v1.5.1
-      with:
-        java-version: 17
-        java-distribution: temurin
-        maven-version: 3.8.6
+      - name: Setup Maven
+        uses: s4u/setup-maven-action@v1.5.1
+        with:
+          java-version: 17
+          java-distribution: temurin
+          maven-version: 3.8.6
+          notify-failure: false
+          skip-build: false
+          expect-tests: false
+          submodules: 'recursive'
 
-    - name: Clean
-      run: ./build.sh clean
-      working-directory: ./build
+      - name: Clean
+        run: ./build.sh clean
+        working-directory: ./build
 
-    - name: Test (commitStage)
-      run: ./build.sh commitStage
-      # run: ./build.sh integrationStage
-      working-directory: ./build
+      - name: Test (commitStage)
+        run: ./build.sh commitStage
+        # run: ./build.sh integrationStage
+        working-directory: ./build
 
       # https://github.com/marketplace/actions/junit-report-action
-    - name: Publish Test Report
-      uses: mikepenz/action-junit-report@v3
-      if: always() # always run even if the previous step fails
-      with:
-        report_paths: 'build/target/testReports/*.xml'
+      - name: Publish Test Report
+        uses: mikepenz/action-junit-report@v3
+        if: always() # always run even if the previous step fails
+        with:
+          report_paths: 'build/target/testReports/*.xml'
 
+  build_master:
```

Comment:
I would be easier to have a separate workflow for what is to be done when something is pushed to master.

Right now this job looks much like a copy of the other job, but it should include everything that is currently done in the [Jenkins task](https://builds.wetransform.to/job/hale/job/hale%7Epublish(master)/).